kafka/consumer_test.go

```diff
@@ -593,7 +599,11 @@ func TestMultipleConsumerGroups(t *testing.T) {
 		m[gid] = &counter
 		cfg.Processor = apmqueue.ProcessorFunc(func(_ context.Context, r apmqueue.Record) error {
 			counter.Add(1)
-			assert.Equal(t, event, r)
+			assert.Equal(t, event.Topic, r.Topic)
+			assert.Equal(t, event.Value, r.Value)
+			// We cannot know which partition the record will be polled from,
+			// but we can assert it's in the range of number of partitions.
+			assert.True(t, event.Partition < partitions-1)
```

Comment:
Same comment on if it's possible for this to ever be false.

Reply:
Yes in case the field is reset from someone modifying the Record during `Process`, since the field is exported.
We could un-export and use a getter, at that point the test can be removed.